pkg/controllers/controlplanemachinesetgenerator/controller_test.go

```diff
@@ -1296,3 +1296,574 @@ var _ = Describe("controlplanemachinesetgenerator controller on Azure", func() {
+
+		usEast1bFailureDomainBuilderGCP = resourcebuilder.GCPFailureDomain().WithZone("us-east-1b")
+
+		usEast1cFailureDomainBuilderGCP = resourcebuilder.GCPFailureDomain().WithZone("us-east-1c")
+
+		usEast1dFailureDomainBuilderGCP = resourcebuilder.GCPFailureDomain().WithZone("us-east-1d")
+
+		usEast1eFailureDomainBuilderGCP = resourcebuilder.GCPFailureDomain().WithZone("us-east-1e")
+
+		usEast1aProviderSpecBuilderGCP = resourcebuilder.GCPProviderSpec().WithZone("us-east-1a")
+
+		usEast1bProviderSpecBuilderGCP = resourcebuilder.GCPProviderSpec().WithZone("us-east-1b")
+
+		usEast1cProviderSpecBuilderGCP = resourcebuilder.GCPProviderSpec().WithZone("us-east-1c")
+
+		usEast1dProviderSpecBuilderGCP = resourcebuilder.GCPProviderSpec().WithZone("us-east-1d")
+
+		usEast1eProviderSpecBuilderGCP = resourcebuilder.GCPProviderSpec().WithZone("us-east-1e")
+
+		cpms3FailureDomainsBuilderGCP = resourcebuilder.GCPFailureDomains().WithFailureDomainBuilders(
+			usEast1aFailureDomainBuilderGCP,
+			usEast1bFailureDomainBuilderGCP,
+			usEast1cFailureDomainBuilderGCP,
+		)
+
+		cpms5FailureDomainsBuilderGCP = resourcebuilder.GCPFailureDomains().WithFailureDomainBuilders(
+			usEast1aFailureDomainBuilderGCP,
+			usEast1bFailureDomainBuilderGCP,
+			usEast1cFailureDomainBuilderGCP,
+			usEast1dFailureDomainBuilderGCP,
+			usEast1eFailureDomainBuilderGCP,
+		)
+
+		cpmsInactive3FDsBuilderGCP = resourcebuilder.ControlPlaneMachineSet().
+						WithState(machinev1.ControlPlaneMachineSetStateInactive).
+						WithMachineTemplateBuilder(
+				resourcebuilder.OpenShiftMachineV1Beta1Template().
+					WithProviderSpecBuilder(
+						usEast1aProviderSpecBuilderGCP.WithZone("").WithMachineType("n1-standard-4"),
+					).
+					WithFailureDomainsBuilder(resourcebuilder.GCPFailureDomains().WithFailureDomainBuilders(
+						usEast1aFailureDomainBuilderGCP,
+						usEast1bFailureDomainBuilderGCP,
+						usEast1cFailureDomainBuilderGCP,
+					)),
+			)
+
+		cpmsInactive5FDsBuilderGCP = resourcebuilder.ControlPlaneMachineSet().
+						WithState(machinev1.ControlPlaneMachineSetStateInactive).
+						WithMachineTemplateBuilder(
+				resourcebuilder.OpenShiftMachineV1Beta1Template().
+					WithProviderSpecBuilder(
+						usEast1aProviderSpecBuilderGCP.WithZone("").WithMachineType("n1-standard-4"),
+					).
+					WithFailureDomainsBuilder(resourcebuilder.GCPFailureDomains().WithFailureDomainBuilders(
+						usEast1aFailureDomainBuilderGCP,
+						usEast1bFailureDomainBuilderGCP,
+						usEast1cFailureDomainBuilderGCP,
+						usEast1dFailureDomainBuilderGCP,
+						usEast1eFailureDomainBuilderGCP,
+					)),
+			)
+
+		cpmsActiveOutdatedBuilderGCP = resourcebuilder.ControlPlaneMachineSet().
+						WithState(machinev1.ControlPlaneMachineSetStateActive).
+						WithMachineTemplateBuilder(
+				resourcebuilder.OpenShiftMachineV1Beta1Template().
+					WithProviderSpecBuilder(
+						usEast1aProviderSpecBuilderGCP.WithZone("").WithMachineType("n1-standard-4"),
```

Comment:
If you look at the other tests (aws/azure), this type is different
```suggestion
						usEast1aProviderSpecBuilderGCP.WithZone("").WithMachineType("n1-standard-8"),
```